src/v/archival/ntp_archiver_service.cc

```diff
@@ -1121,30 +1297,36 @@ ss::future<ntp_archiver::batch_result> ntp_archiver::wait_all_scheduled_uploads(
           return s.upload_kind == segment_upload_kind::non_compacted;
       });
 
+    // Inline manifest upload in regular non-compacted uploads
```

Comment:
why manifest upload in compacted case is not inlined? what is the difference?

Reply:
If there are any non-compacted uploads then we inline the manifest upload with them, otherwise we may inline it with the compacted uploads.  The important thing is just that we only inline with one or the other, not both.